framework/src/headers/ip/v6/mod.rs

```diff
@@ -198,6 +200,7 @@ impl Ipv6Header {
                 // TODO: should validate we are only skipping v6 extension headers
                 while next_hdr != TCP_NXT_HDR
                     && next_hdr != UDP_NXT_HDR
+                    && next_hdr != ICMP_NXT_HDR
```

Comment:
we probably want to do checksum tests specific to icmp as well. Look at packets.rs (in the test folder) as an example. You can extract icmp message bytes from wireshark for example.